store/src/java/com/zimbra/cs/service/mail/ModifyProfileImage.java

```diff
@@ -200,4 +219,38 @@ private int updateDatabase(Mailbox mbox, OperationContext octxt, String imageFol
             MailItem.Type.DOCUMENT, null, customMetaData, false);
         return imgDocItem.getId();
     }
+    
+    private static void removeFromLDAP(Mailbox mbox, ZimbraSoapContext zsc) throws IOException, ServiceException {
+        HashMap<String, Object> prefs = new HashMap<String, Object>();
+        prefs.put(Provisioning.A_thumbnailPhoto, null);
+        Provisioning.getInstance().modifyAttrs(mbox.getAccount(), prefs, true,
+            zsc.getAuthToken());
+        return;
+    }
+    
+    private void removeFromDatabase(Mailbox mbox, OperationContext octxt, String imageFolderName)
+        throws ServiceException {
+        int folderId;
+        try {
+            Folder imgFolder = mbox.getFolderByName(octxt, Mailbox.ID_FOLDER_ROOT, imageFolderName);
+            folderId = imgFolder.getId();
```

Comment:
Could imgFolder be null here - perhaps for old mailboxes that existed before this feature?

Reply:
NO_SUCH_FOLDER is being handled in catch

Reply:
Fair enough, looks like `getFolderByName` can't return null.